modal/cloud_bucket_mount.py

```diff
@@ -135,8 +135,12 @@ def cloud_bucket_mounts_to_proto(mounts: List[Tuple[str, _CloudBucketMount]]) ->
         if mount.requester_pays and not mount.secret:
             raise ValueError("Credentials required in order to use Requester Pays.")
 
-        if not mount.key_prefix.endswith('/'):
+        if mount.key_prefix == "":
+            raise ValueError("key_prefix must not be empty. To mount the entire bucket, do not specify any prefix.")
```

Comment:
I might be barging in after a useful discussion but is there any specific reason to disallow this?